File: AmericanExpress/src/main/java/com/braintreepayments/api/AmericanExpressGetRewardsBalanceCallback.java
package com.braintreepayments.api;

import androidx.annotation.Nullable;

/**
 * Callback for receiving result of {@link AmericanExpressClient#getRewardsBalance(String, String, AmericanExpressGetRewardsBalanceCallback)}.
 */
public interface AmericanExpressGetRewardsBalanceCallback {

    /**
     * @param rewardsBalance {@link AmericanExpressRewardsBalance}
     * @param error an exception that occurred while fetching rewards balance
     */
    void onResult(@Nullable AmericanExpressRewardsBalance rewardsBalance, @Nullable Exception error);
}

File: AmericanExpress/src/main/java/com/braintreepayments/api/AmericanExpressGetRewardsBalanceCallback.kt
package com.braintreepayments.api

/**
 * Callback for receiving result of
 * [AmericanExpressClient.getRewardsBalance].
 */
fun interface AmericanExpressGetRewardsBalanceCallback {
    /**
     * @param americanExpressResult the [AmericanExpressResult]
     */
    fun onAmericanExpressResult(americanExpressResult: AmericanExpressResult?)
}